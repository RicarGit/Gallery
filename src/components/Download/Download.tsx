import * as S from './Download.styled'
import downloadIcon from './download.svg'
import { api } from 'services/api'

export type DownloadImage = {
  url: string
  name: string
}

export const Download = ({ url, name }: DownloadImage) => {
  return (
    <S.Container>
      <div className='download' onClick={() => api.downloadImage(url, name)}>
        <img src={downloadIcon} alt='download icon' ></img>
      </div>
    </S.Container>
  )
}