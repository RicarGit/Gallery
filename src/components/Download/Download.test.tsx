import { render } from '@testing-library/react'
import { Download } from './Download'

describe('Download', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Download url='http://www.google.com' name='image name' />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})