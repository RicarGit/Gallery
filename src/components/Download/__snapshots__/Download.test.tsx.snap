// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Download should match snapshot 1`] = `
<div
  class="Downloadstyled__Container-sc-9k28x7-0 ezUDzu"
>
  <div
    class="download"
  >
    <img
      alt="download icon"
      src="download.svg"
    />
  </div>
</div>
`;
